h2o-genmodel/src/main/java/hex/genmodel/CategoricalEncoding.java

```diff
@@ -4,76 +4,12 @@
 
 import java.util.Map;
 
-public enum CategoricalEncoding {
-  AUTO {
-    @Override
-    public Map<String, Integer> createColumnMapping(GenModel m) {
-      return new EnumEncoderColumnMapper(m).create();
-    }
-
-    @Override
-    public Map<Integer, CategoricalEncoder> createCategoricalEncoders(GenModel m, Map<String, Integer> columnMapping) {
-      return new EnumEncoderDomainMapConstructor(m, columnMapping).create();
-    }
-  },
-  OneHotExplicit {
-    @Override
-    public Map<String, Integer> createColumnMapping(GenModel m) {
-      return new OneHotEncoderColumnMapper(m).create();
-    }
-
-    @Override
-    public Map<Integer, CategoricalEncoder> createCategoricalEncoders(GenModel m, Map<String, Integer> columnMapping) {
-      return new OneHotEncoderDomainMapConstructor(m, columnMapping).create();
-    }
-  },
-  Binary {
-    @Override
-    public Map<String, Integer> createColumnMapping(GenModel m) {
-      return new BinaryColumnMapper(m).create();
-    }
-
-    @Override
-    public Map<Integer, CategoricalEncoder> createCategoricalEncoders(GenModel m, Map<String, Integer> columnMapping) {
-      return new BinaryDomainMapConstructor(m, columnMapping).create();
-    }
-  },
-  EnumLimited {
-    @Override
-    public Map<String, Integer> createColumnMapping(GenModel m) {
-      return new EnumLimitedEncoderColumnMapper(m).create();
-    }
-
-    @Override
-    public Map<Integer, CategoricalEncoder> createCategoricalEncoders(GenModel m, Map<String, Integer> columnMapping) {
-      return new EnumLimitedEncoderDomainMapConstructor(m, columnMapping).create();
-    }
-  },
-  Eigen {
-    @Override
-    public Map<String, Integer> createColumnMapping(GenModel m) {
-      return new EigenEncoderColumnMapper(m).create();
-    }
-
-    @Override
-    public Map<Integer, CategoricalEncoder> createCategoricalEncoders(GenModel m, Map<String, Integer> columnMapping) {
-      return new EigenEncoderDomainMapConstructor(m, columnMapping).create();
-    }
-  },
-  LabelEncoder {
-    @Override
-    public Map<String, Integer> createColumnMapping(GenModel m) {
-      return new EnumEncoderColumnMapper(m).create();
-    }
+public interface CategoricalEncoding {
```

Comment:
so this refactor to interface and move of the enums is entirely unnecessary for now? if yes I would lean towards reverting this part